src/core/config.go

```diff
@@ -755,9 +757,11 @@ func (config *Configuration) Hash() []byte {
 	for _, l := range config.Licences.Reject {
 		h.Write([]byte(l))
 	}
-	for _, env := range config.getBuildEnv(false, false) {
-		if !strings.HasPrefix(env, "SECRET") {
-			h.Write([]byte(env))
+	env := config.getBuildEnv(false, false)
+	for _, k := range slices.Sorted(maps.Keys(env)) {
+		if !strings.HasPrefix(k, "SECRET") {
+			h.Write([]byte(k))
+			h.Write([]byte(env[k]))
```

Comment:
Maybe have a delimiter between them? There's a degenerate case in which two environments could still produce the same hash this way (e.g. `TEST=TEST`, `TES=TTEST`). `=` is probably the safest bet.

Reply:
Yup, makes sense!